core/src/main/java/org/apache/iceberg/SchemaUpdate.java

```diff
@@ -382,27 +378,43 @@ private int assignNewColumnId() {
   }
 
   private TableMetadata applyChangesToMapping(TableMetadata metadata) {
-    String mappingJson = metadata.property(TableProperties.DEFAULT_NAME_MAPPING, null);
```

Comment:
Do we need to rename `mappingJson`? Seems like it is optional so may be a good idea to avoid unnecessary changes to simplify cherry-picks and reviews.

Reply:
I'd still skip the rename if possible.